tests/unit_tests/db_engine_specs/test_drill.py

```diff
@@ -64,4 +64,4 @@ def test_sadrill_impersonation(app_context: AppContext) -> None:
     url = URL("drill+sadrill")
     username = "DoAsUser"
     DrillEngineSpec.modify_url_for_impersonation(url, True, username)
-    assert url.username == username
+    assert url.query["impersonation_target"] == username
```

Comment:
nit: bonus points for adding impersonation tests for `drill+odbc`, `drill+jdbc` and `drill+foobar`.

Reply:
The first two cases are already higher up in test_drill.py.  I've now added the drill+foobar case.

Reply:
Apologies, I stand corrected!